lamindb/models/_django.py

```diff
@@ -18,6 +18,30 @@
     from .record import Record
 
 
+def patch_many_to_many_descriptor() -> None:
+    """Patches Django's `ManyToManyDescriptor.__get__` method to suggest better errors when saving relationships of an unsaved model.
+
+    Before this patch: Cryptic errors are raised when relationships of an unsaved
+    record are attempted to be modified.
+
+    After this patch: Attempts to access M2M relationships on unsaved objects
+    will raise ValueError, suggesting explicit .save() of the record to be modified
+    before relationship creation.
+    """
+    from django.db.models.fields.related_descriptors import ManyToManyDescriptor
+
+    original_add = ManyToManyDescriptor.__get__
+
+    def patched_get(self, instance, cls=None):
+        if instance is not None and instance.pk is None:
+            raise ValueError(
```

Comment:
I think it's a bit confusing to assume users are assigning the relationships when they are just accessing them.

Also we need to spell out many-to-many relationships like the original message, because FK relationships don't raise this error:
<img width="554" alt="Screenshot 2025-05-12 at 10 26 30" src="https://github.com/user-attachments/assets/a4cba6ec-1bcf-40df-98a0-cee9b82b9f5b" />

Reply:
I agree, the problem is that it is triggered on any access.